### PR TITLE
Explain --domains and --cert-name usage

### PR DESCRIPTION
**Pull request by bmw**
Supersedes #4021.

Updated in response to Peter's changes on my previous PR and rebased into individual commits by author to preserve authorship. This should be merged and not squashed.

### CODE REVIEW

File: certbot/cli.py [hunk 1]
```diff
@@ -867,14 +867,20 @@ def prepare_and_parse_args(plugins, args, detect_defaults=False):  # pylint: dis
         metavar="DOMAIN", action=_DomainsAction, default=[],
         help="Domain names to apply. For multiple domains you can use "
              "multiple -d flags or enter a comma separated list of domains "
-             "as a parameter. (default: Ask)")
+             "as a parameter. The first provided domain will be used in "
+             "filenames and some software user interfaces unless otherwise "
```
Comment (SwartzCr) [inline certbot/cli.py]: what filenames?
I find this a little too ambiguous to be comfortable with it coming out of --help

Comment (bmw) [inline certbot/cli.py]: What would you like to see instead?

Comment (SwartzCr) [inline certbot/cli.py]: maybe "in filenames used for storing the certificate and key material" or something?

**Comment by bmw** [comment]
@SwartzCr: ping

**Pull request merged**